workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs

```diff
@@ -4,374 +4,431 @@
-                }
-                else
-                {
-                    PlaceTile(tileCoord);
-                }
+                return false;
             }
 
-            if (tileCoord.x == tileCoord.y ||
-                tileCoord.x < 0 && tileCoord.x == -tileCoord.y ||
-                tileCoord.x > 0 && tileCoord.x == 1 - tileCoord.y)
+            if (!TryLoadResource(GroundEdgePath, out groundEdge))
             {
-                diff = new Vector2Int(-diff.y, diff.x);
+                return false;
             }
 
-            tileCoord += diff;
-        }
-
-        tileParentTransform.position = new Vector3()
-        {
-            x = tileParentTransform.position.x,
-            y = HeightOffset,
-            z = tileParentTransform.position.z
-        };
-    }
+            if (!TryLoadResource(SurroundPath, out surroundWall))
+            {
+                return false;
+            }
 
-    private void PlaceTile(Vector2Int tileCoord)
-    {
-        if (Random.value < EmptyTileChance)
-        {
-            return;
-        }
+            if (!TryLoadResource(CornerPath, out cornerPiece))
+            {
+                return false;
+            }
 
-        var tile = levelTiles[Random.Range(0, levelTiles.Length)];
-        float rotation = 90 * Random.Range(0, 4);
+            centreTiles = new[]
+            {
+                Resources.Load<GameObject>(CentreTile0),
+                Resources.Load<GameObject>(CentreTile1),
+                Resources.Load<GameObject>(CentreTile2),
+                Resources.Load<GameObject>(CentreTile3)
+            };
 
-        PlaceTile(tileCoord, tile, rotation);
-    }
+            if (centreTiles.Any(t => t == null))
+            {
+                Debug.LogError("Failed to load CentreTile resource (Expecting all the following paths to exist: " +
+                    $"\n\tResources/{CentreTile0}... (Click to expand)" +
+                    $"\n\tResources/{CentreTile1}" +
+                    $"\n\tResources/{CentreTile2}" +
+                    $"\n\tResources/{CentreTile3}");
+                return false;
+            }
 
-    private void PlaceTile(Vector2Int tileCoord, GameObject tile, float rotation)
-    {
-        var tileOffset = TileSeparation / 2;
+            levelTiles = Resources.LoadAll<GameObject>(LevelTilePath);
 
-        Instantiate(
-            tile,
-            new Vector3()
-            {
-                x = (tileCoord.x - 1) * TileSeparation + tileOffset,
-                z = (tileCoord.y - 1) * TileSeparation + tileOffset
-            },
-            new Quaternion()
+            if (levelTiles.Length <= 0)
             {
-                eulerAngles = new Vector3()
-                {
-                    y = rotation
-                }
-            },
-            tileParentTransform.transform);
-    }
+                Debug.LogError($"Failed to load resource at Resources/{LevelTilePath}");
```

Comment:
This error message doesn't really correspond to the `if` statement ?

Reply:
I don't follow this one, Jamie...the `if` statement checks if there were resources found, and if not, prints the error message?

Do you mean it ought to read, `Failed to load any resources at Resources/{LevelTilePath}`?

Reply:
Oh I completely missed the statement above this one! Disregard this comment 😬